impl/sql/mutationstorage/mutation_logs.go

```diff
@@ -202,6 +206,9 @@ func (m *Mutations) HighWatermark(ctx context.Context, directoryID string, logID
 // ReadLog may return more rows than batchSize in order to fetch all the rows at a particular timestamp.
 func (m *Mutations) ReadLog(ctx context.Context, directoryID string,
 	logID, low, high int64, batchSize int32) ([]*mutator.LogMessage, error) {
+	if high > watermark(time.Now().Add(time.Minute)) {
```

Comment:
This feels really arbitrary. Locking the code to time.Now() to check types feels like a medicine that has some unwelcome side-effects.